File: sensors/test/read_gpio.py
#!/usr/bin/env python

# Read from BCM pin with pulldown active

import RPi.GPIO as GPIO
import sys

PIN = int(sys.argv[1])

GPIO.setmode(GPIO.BCM)
GPIO.setup(PIN, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)

while True:
    state = GPIO.input(PIN)
    print(state)

File: sensors/test/read_gpio.c
// Read from GPIO 9 (wiringPi 13)
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

int main(int argc, char** argv) {
  if (wiringPiSetupGpio() == -1)
    exit(1);

  int pin = atoi(argv[1]);
  pinMode(pin, INPUT);
  pullUpDnControl(pin, PUD_DOWN);

  printf("pin %d configured %d\n", pin, PUD_DOWN);

  while (1) {
    printf("%d\n", digitalRead(pin));
  }
}

